src/main/java/seedu/address/ui/TodayPanel.java

```diff
@@ -0,0 +1,126 @@
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.StackPane;
+import seedu.address.commons.util.DateUtil;
+import seedu.address.model.ReadOnlyProjectsFolder;
+import seedu.address.model.project.Project;
+import seedu.address.model.task.CompletableDeadline;
+import seedu.address.model.task.repeatable.Event;
+
+/**
+ * Panel displaying today screen.
+ */
+public class TodayPanel extends UiPart<Region> {
+    private static final String FXML = "TodayPanel.fxml";
+
+    private final LocalDate currentDate;
+    private final ObservableList<Project> projectsList;
+
+    private final ListView<Event> eventsListView = new ListView<>();
+    private final ListView<CompletableDeadline> deadlinesListView = new ListView<>();
+
+    private Label noEventsPlaceholder;
+    private Label noDeadlinesPlaceholder;
+
+    @FXML
+    private Label date;
+
+    @FXML
+    private StackPane eventsListViewPlaceholder;
+
+    @FXML
+    private StackPane deadlinesListViewPlaceholder;
+
+    /**
+     * Creates a {@code TodayPanel}.
+     * @param projectsFolder Projects folder used to create today panel.
+     */
+    public TodayPanel(ReadOnlyProjectsFolder projectsFolder, LocalDate date) {
+        super(FXML);
+
+        this.projectsList = projectsFolder.getProjectsList();
+        this.currentDate = date;
+        this.date.setText(DateUtil.decodeDate(currentDate));
+
+        initEventsSection();
+        initDeadlinesSection();
+    }
+
+    private void initDeadlinesSection() {
+        ObservableList<CompletableDeadline> deadlinesListViewItems = deadlinesListView.getItems();
+        deadlinesListView.setCellFactory(listView -> new CompletableDeadlineListViewCell());
+        for (Project project : projectsList) {
+            FilteredList<CompletableDeadline> deadlineList = project.getDeadlinesOnDate(currentDate);
+            deadlinesListViewItems.addAll(deadlineList);
+        }
+
+        if (deadlinesListViewItems.isEmpty()) {
+            noDeadlinesPlaceholder = new Label();
+            noDeadlinesPlaceholder.setText("You have no deadlines today!");
+            deadlinesListViewPlaceholder.getChildren().add(noDeadlinesPlaceholder);
+        } else {
+            deadlinesListViewPlaceholder.getChildren().add(deadlinesListView);
+        }
+    }
+
+    private void initEventsSection() {
+        ObservableList<Event> eventsListViewItems = eventsListView.getItems();
+        eventsListView.setCellFactory(listView -> new EventListViewCell());
+        for (Project project : projectsList) {
+            FilteredList<Event> eventList = project.getEventsOnDate(currentDate);
+            eventsListViewItems.addAll(eventList);
+        }
+
+        if (eventsListViewItems.isEmpty()) {
+            noEventsPlaceholder = new Label();
+            noEventsPlaceholder.setText("You have no events today!");
```

Comment:
Wanna change the string text for this and deadlines to a constant declared at the top for easy access?

Reply:
sure